js/views/conversation_view.js

```diff
@@ -241,6 +278,10 @@
             this.model.messageCollection.add(message, {merge: true});
             message.setToExpire();
 
+            if (this.lastSeenIndicator) {
+                this.lastSeenIndicator.increment(1);
+            }
+
```

Comment:
One thing that helps us is that we reset every time a conversation is opened. So we're talking about a conversation which is open and had unread messages when we opened it. And this is inside the `addMessage` function, bound to the `newmessage` event on the `Conversation` model.

It comes down to whether the `newmessage` event can be trusted for alerting us to every incoming message. Looking through the code, I see situations where other kinds of incoming data (like key change notifications) can trigger this, so the 'unread messages' text may not be fully appropriate in that case. Thoughts?